docs/trace/building-your-own-processor/Program.cs

```diff
@@ -27,10 +27,12 @@ public static void Main()
     {
         using var otel = Sdk.CreateTracerProvider(b => b
             .AddActivitySource("MyCompany.MyProduct.MyLibrary")
-            .AddProcessorPipeline(pipeline =>
-            {
-                pipeline.AddProcessor(current => new MyActivityProcessor());
-            }));
+
+            // TODO: seems buggy if you remove A and B here, MyActivityProcessor(C).OnEnd is not called.
+            // TODO: should the dispose order be C, B, A or A, B C?
+            .AddProcessorPipeline(p => p.AddProcessor(current => new MyActivityProcessor("A")))

```

Comment:
This is adding multiple processor pipelines. I guess you wanted to add multiple processors to the same, single pipeline?

Reply:
Nope, I was using this to showcase how pipeline could be confusing, and I think we need to change it.

Reply:
The fan out scenario is a separate topic, and the name should be changed #979.